arrays/arrays.c

```diff
@@ -83,14 +99,22 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
-
+  if (index > arr->count) {
+    fprintf(stderr, "INSERT error: %d", index);
+  }
   // Resize the array if the number of elements is over capacity
-
+  if (arr->count + 1 > arr->capacity) { 
+    resize_array(arr);
+  }
   // Move every element after the insert index to the right one position
-
+  for(int i = index; i < arr->count; i++) {
+    arr->elements[i+1] = arr->elements[i];
+  }
   // Copy the element and add it to the array
-
+  char *copy = strdup(element);
+  arr->elements[arr->count] = copy;
```

Comment:
You're inserting your element at the last index in the array when you use [arr->count] here. I think you meant to insert the element at the passed in index from line 99, so your term in the square brackets needs to change.